frontend/src/scenes/setup/ToolbarSettings.js

```diff
@@ -0,0 +1,44 @@
+import React, { useState } from 'react'
+import { useValues, useActions } from 'kea'
+import { userLogic } from '../userLogic'
+import { Switch } from 'antd'
+
+export function ToolbarSettings() {
+    const { user } = useValues(userLogic)
+    const { userUpdateRequest } = useActions(userLogic)
+    const [saved, setSaved] = useState(false)
+
+    return (
+        <div>
+            <Switch
+                onChange={() => {
+                    userUpdateRequest({
+                        user: {
+                            toolbar_mode: user.toolbar_mode === 'toolbar' ? 'default' : 'toolbar',
+                        },
+                    })
+                    setSaved(true)
+                }}
+                defaultChecked={user.toolbar_mode === 'toolbar'}
+            />
+            <label
+                style={{
+                    marginLeft: '10px',
+                }}
+            >
+                Try the new PostHog Toolbar?
+                {saved && (
+                    <span className="text-success" style={{ marginLeft: 10 }}>
+                        Preference saved. You might need to restart your browser session for the change to take effect.
```

Comment:
I didn't understand "restart browser session" -- do you mean "restart my browser" or just refresh the page?

Either way when I went to 127.0.0.1:8000/demo it worked

Reply:
The first option - restarting the browser. With the current posthog-js, if you open the toolbar through "posthog -> actions -> inspect element on site", it'll jam the toolbar config as it was at that moment into `window.sessionStorage`. This includes information on which version of the toolbar to load.

Later if you visit your site, the data is taken from there... no matter what /decide replies. If you then use the toggle to use the new toolbar, it won't show up... as the data in sessionStorage will still force the old one to open.

This can be cleaned up, but that requires a new version of posthog-js. It's something I already started working on, but I want to get the tests working first (in progress).

Reply:
That makes sense, it'd be good to change the language to "restart your browser" which would be less confusing (for me at least!)

Reply:
Fixed!